pkg/client/client.go

```diff
@@ -344,3 +424,29 @@ func monitorKeepTerminatedEvent(
 
 	logger.Info("unsubscribing from events on keep terminated")
 }
+
+// waitForEventConfirmation ensures that after receiving specific number of block
+// confirmations the state of the chain is actually as expected. It waits for
+// predefined number of blocks since the start block number provided. After the
+// required block number is reached it performs a check of the chain state with
+// a provided function returning a boolean value.
+func waitForEventConfirmation(
+	ethereumChain eth.Handle,
+	startBlockNumber uint64,
+	stateCheck func() (bool, error),
+) (bool, error) {
+	blockHeight := startBlockNumber + blockConfirmations
+	logger.Infof("waiting for [%d] block", blockHeight)
```

Comment:
It looks weird, especially that we say `... at block [x]` in other places:
`21:53:13.777  INFO keep-ecdsa: waiting for [5594] block client.go:439` 

Can we change it to `"waiting for block [%d]"`?

Reply:
https://github.com/keep-network/keep-ecdsa/pull/374/commits/a59937b1f525f62015d645cb9c746ad83441d93f